proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+
+## Architecture at a Glance
+
+### Quick Comparison
+
+| Aspect                 | Current (S3-compatible Storage)                 | Proposed (Filesystem Storage)                     |
+|------------------------|-------------------------------------------------|---------------------------------------------------|
+| **Storage Backend**    | S3-compatible (SeaweedFS, MinIO, S3, GCS, etc.) | Kubernetes PVC                                    |
+| **URI Scheme**         | `s3://`, `gs://`, `minio://`                    | `kfp-artifacts://`                                |
+| **Architecture**       | Separate object storage service                 | KFP-native artifact server                        |
+| **Required Knowledge** | S3 concepts (buckets, endpoints, regions)       | Kubernetes concepts (PVCs, StorageClasses)        |
+| **Multi-tenancy**      | Shared storage (single instance)                | Per-namespace PVCs (in namespace-local mode)      |
+| **Scaling**            | Single object storage instance                  | Per-namespace artifact servers and storage quotas |
+
+### High-Level Flow
+
+```text
+Pipeline Submission
+        │
+        ▼
+Driver (detects "kfp-artifacts://")
+        │
+        ▼
+Artifact Server (mounts PVC)
+        │
+        ├──────────────────────────────────┐
+        │                                  │
+        ▼                                  ▼
+Pipeline Pods                         UI/API Clients
+(upload/download via API)             (view artifacts)
+        │                                  │
+        └──────────► Artifact Server ◄─────┘
+                           │
+                           ▼
+                    Read/Write to PVC
+```
+
+### Key Components
+
+- **Storage**: Kubernetes `PersistentVolumeClaims` (one per namespace)
+- **URI Format**: `kfp-artifacts://<namespace>/<pipeline>/<run-id>/<node-id>/<artifact-name>`
+- **API Changes**: New `/filesystem-storage/config` endpoint for UI configuration discovery
+- **Deployment Modes**:
+  - **Central**: Single artifact server for all namespaces (default)
+  - **Namespace-local**: One artifact server per namespace (better isolation)
+  - **Mixed**: Per-namespace mode override via `kfp-launcher` ConfigMap
+
+## Motivation
+
+**This proposal does not aim to replace existing object storage solutions.** S3-compatible storage remains fully supported and recommended for most production workloads. Instead, this KEP provides an additional option for deployments where a simpler, KFP-native artifact storage solution is preferred.
+
+While KFP currently ships with S3-compatible storage by default, this still requires deploying and maintaining a separate object storage service. For some deployment scenarios, this additional component may not be desired.
+
+### Reduced External Dependencies
```

Comment:
I don't think this section adds a lot and could be consolidated in the Motivation section.